js/apps/admin-ui/src/clients/authorization/PermissionDetails.tsx

```diff
@@ -273,24 +249,15 @@ export default function PermissionDetails() {
               />
             </FormGroup>
             {applyToResourceTypeFlag ? (
-              <FormGroup
+              <TextControl
+                name="resourceType"
                 label={t("resourceType")}
-                fieldId="resourceType"
-                labelIcon={
-                  <HelpItem
-                    helpText={t("resourceTypeHelp")}
-                    fieldLabelId="resourceType"
-                  />
-                }
-                isRequired={permissionType === "scope"}
-              >
-                <KeycloakTextInput
-                  id="resourceType"
-                  {...register("resourceType", {
-                    required: permissionType === "scope",
-                  })}
-                />
-              </FormGroup>
+                labelIcon={t("resourceTypeHelp")}
+                rules={{
+                  required:
+                    permissionType === "scope" ? t("required") : undefined,
```

Comment:
You need to add a message here
```suggestion
                  required: { value:
                       permissionType === "scope" ? t("required") : undefined,
                       message: t("required"),
```